docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2422,3 +2236,189 @@ recommend at least one test for your core workflows that follows this pattern:
 3. Upload version 2.0 of your DAR.
 4. Switch your backends to start using version 2.0, ideally this should be a flag.
 5. Validate that the core workflows are in the same state and advance them to check that they are not stuck.
+
+SCU Support in Daml Tooling
+===========================
+
+Codegen
+-------
+
+For packages that support SCU (i.e. LF1.17), generated code uses
+package-names in place of package-ids in template IDs. Retrieved data
+from the ledger is subject to the upgrade transformations described
+in previous sections.
+
+Concretely, this is implemented as follows:
+
+Java
+~~~~
+
+The classes that are generated for each template and interface contain a
+``TEMPLATE_ID`` field, which, for upgradable packages, now use a
+package name rather than a package ID. To help you determine
+the package ID of these packages, we have added a new ``PACKAGE_ID`` field to all
+such classes. Upgradable packages also cause ``PACKAGE_NAME`` and
+``PACKAGE_VERSION`` fields to be present.
```

Comment:
```suggestion
The classes that are generated for each template and interface contain a
``TEMPLATE_ID`` field, which, for upgradable packages, now use a
package name rather than a package ID. To help you determine
the package ID of these packages, we have added a new ``PACKAGE_ID`` field to all
such classes. Upgradable packages also have ``PACKAGE_NAME`` and
``PACKAGE_VERSION`` fields.
```